Unable to set 'session.gc_maxlifetime'

In a clean install with default config, when I click on the "Add client" button I get this error
```
SimpleSAML_Error_Error: UNHANDLEDEXCEPTION

Backtrace:
1 www/_include.php:45 (SimpleSAML_exception_handler)
0 [builtin] (N/A)
Caused by: Twig_Error_Runtime: An exception has been thrown during the rendering of a template ("Unable to set 'session.gc_maxlifetime' to value '10800' when session has been started by session.auto_start or session_start().").
Backtrace:
20 modules/oidc/templates/clients/_form.twig:65 (Twig_Template::displayBlock)
19 vendor/twig/twig/lib/Twig/Environment.php(467) : eval()'d code:82 (__TwigTemplate_8169d88dedf1496eb3227be4fc154e11fd5a804a53bc2bc6208f196779ab8742::block_contentwrapper)
18 vendor/twig/twig/lib/Twig/Template.php:215 (Twig_Template::displayBlock)
17 vendor/twig/twig/lib/Twig/Environment.php(467) : eval()'d code:76 (__TwigTemplate_5fdce0d453774ad901c3fee798ae80a56c308da06432609ef01c44625b3a2c4f::doDisplay)
16 vendor/twig/twig/lib/Twig/Template.php:432 (Twig_Template::displayWithErrorHandling)
15 vendor/twig/twig/lib/Twig/Template.php:403 (Twig_Template::display)
14 vendor/twig/twig/lib/Twig/Environment.php(467) : eval()'d code:30 (__TwigTemplate_8169d88dedf1496eb3227be4fc154e11fd5a804a53bc2bc6208f196779ab8742::doDisplay)
13 vendor/twig/twig/lib/Twig/Template.php:432 (Twig_Template::displayWithErrorHandling)
12 vendor/twig/twig/lib/Twig/Template.php:403 (Twig_Template::display)
11 vendor/twig/twig/lib/Twig/Environment.php(467) : eval()'d code:26 (__TwigTemplate_a615e0cc71f506d83bf129a67305a70d7c164e79f1d0a814f56a6a965cbebb56::doDisplay)
10 vendor/twig/twig/lib/Twig/Template.php:432 (Twig_Template::displayWithErrorHandling)
9 vendor/twig/twig/lib/Twig/Template.php:403 (Twig_Template::display)
8 vendor/twig/twig/lib/Twig/Environment.php(467) : eval()'d code:27 (__TwigTemplate_3cf4dc4f4f8444ec078bb8a42c43daa1745f0c9c0dfc6c52fcdb841e4e2a694e::doDisplay)
7 vendor/twig/twig/lib/Twig/Template.php:432 (Twig_Template::displayWithErrorHandling)
6 vendor/twig/twig/lib/Twig/Template.php:403 (Twig_Template::display)
5 vendor/twig/twig/lib/Twig/Template.php:411 (Twig_Template::render)
4 vendor/twig/twig/lib/Twig/Environment.php:363 (Twig_Environment::render)
3 lib/SimpleSAML/XHTML/Template.php:406 (SimpleSAML_XHTML_Template::show)
2 modules/oidc/lib/Services/RoutingService.php:49 (SimpleSAML\Modules\OpenIDConnect\Services\RoutingService::call)
1 modules/oidc/www/clients/new.php:18 (require)
0 www/module.php:135 (N/A)
```

Environment
- centos 7
- mariadb100u
- simplesamlphp 1.15.4

--- a/lib/Form/ClientForm.php
+++ b/lib/Form/ClientForm.php
@@ -35,7 +35,7 @@ class ClientForm extends Form
         $this->onValidate[] = [$this, 'validateRedirectUri'];
 
         $this->setMethod('POST');
-        $this->addProtection('Security token has expired, please submit the form again');
+        $this->addComponent(new Controls\CsrfProtection(null), Form::PROTECTOR_ID);
 
         $this->addText('name', '{oidc:client:name}')
             ->setMaxLength(255)

--- a/lib/Form/Controls/CsrfProtection.php
+++ b/lib/Form/Controls/CsrfProtection.php
@@ -1,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the simplesamlphp-module-oidc.
+ *
+ * Copyright (C) 2018 by the Spanish Research and Academic Network.
+ *
+ * This code was developed by Universidad de Córdoba (UCO https://www.uco.es)
+ * for the RedIRIS SIR service (SIR: http://www.rediris.es/sir)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SimpleSAML\Modules\OpenIDConnect\Form\Controls;
+
+use Nette\Forms\Controls\CsrfProtection as BaseCsrfProtection;
+use SimpleSAML\Utils\Random;
+
+class CsrfProtection extends BaseCsrfProtection
+{
+    public function __construct($errorMessage)
+    {
+        parent::__construct($errorMessage);
+
+        $this->session = \SimpleSAML_Session::getSession();
+    }
+
+    public function getToken()
+    {
+        $token = $this->session->getData('token', 'form');
+
+        if (!isset($token)) {
+            $token = Random::generateID();
+
+            $this->session->setData('token', 'form', $token);
+        }
+
+        return $token;
+    }
+}

